plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/price/block.tsx

```diff
@@ -34,11 +42,43 @@ interface PriceProps {
 	price_range: null | { min_amount: string; max_amount: string };
 }
 
+/**
+ * Converts a price string from the Admin API format to the Store API format.
+ *
+ * The Admin API returns prices as decimal numbers (e.g., "12.99"), while the Store API
+ * expects prices as integers representing the smallest currency unit (e.g., "1299" for $12.99).
+ * This function multiplies the price by 100 to convert from decimal to minor units.
+ *
+ * @param priceString The price as a string from the Admin API (e.g., "12.99")
+ * @param fallback    The fallback value if priceString is null/undefined (defaults to "0")
+ * @return The price converted to minor units as a string (e.g., "1299")
+ */
+const convertAdminPriceToStoreApiFormat = (
+	priceString: string | null | undefined,
+	fallback = '0'
+) => {
+	return ( Number.parseFloat( priceString ?? fallback ) * 100 ).toString();
```

Comment:
It's incorrect for some of the currencies. Not all currencies have two decimal points, sometimes it's 0 or 3. You should use `minorUnit` for it:

```suggestion
    const multiplier = 10 ** minorUnit;
	return ( Number.parseFloat( priceString ?? fallback ) * multiplier ).toString();
```

Reply:
Nice catch! Fixed with e2e4de557dc37266534b21309454691a1c86e83a